clients/roscpp/src/libros/callback_queue.cpp

```diff
@@ -382,14 +393,6 @@ CallbackQueue::CallOneResult CallbackQueue::callOneCB(TLS* tls)
         result = cb->call();
       }
     }
-    catch (std::exception&)
-    {
-      // ensure that thread id gets restored, even in case of an exception
-      tls->calling_in_this_thread = last_calling;
-      throw;
```

Comment:
Sure, the community link provided shows an exception that was thrown by user code but the strack trace in gdb leads to callOneCB. The exception is the same but the source of the exception is lost.